lib/hyacinth/utils/csv_import_export_utils.rb

```diff
@@ -175,6 +183,117 @@ def self.create_import_job_from_csv_data(csv_data_string, import_filename, user)
+
+    identifiers_to_row_numbers = identifiers_to_row_numbers_for_csv_data(csv_data_string)
+
+    # Return if no identifiers were found in the csv data, since that means that processing order doesn't matter
+    return {} if identifiers_to_row_numbers.blank?
+
+    prerequisite_rows_based_on_headers(csv_data_string, row_number_for_header_row, index_of_parent_headers, identifiers_to_row_numbers)
+  end
+
+  def self.prerequisite_rows_based_on_headers(csv_data_string, row_number_for_header_row, index_of_parent_headers, identifiers_to_row_numbers)
+    prerequisite_row_map = {}
+    if index_of_parent_headers.length > 0
+
+      prerequisite_rows_to_dependent_rows = {}
+      row_counter = -1
+      CSV.parse(csv_data_string) do |row|
+        row_counter += 1
+        next if row_counter <= row_number_for_header_row
+
+        ### Note: We can't guarantee that a row with a pid is in Hyacinth already.
+        # It could be an indication that we want to import an object that's in Fedora,
+        # but not Hyacinth.
+
+        index_of_parent_headers.each do |index|
+          parent_identifier = row[index]
+          next unless identifiers_to_row_numbers.key?(parent_identifier)
+          parent_row_number = identifiers_to_row_numbers[parent_identifier]
+          prerequisite_row_map[row_counter] ||= []
+
+          # If THIS row's prerequisite row has other rows that are dependent on it,
+          # then THIS row's prerequisite should be the HIGHEST one of those rows.
+          if prerequisite_rows_to_dependent_rows.key?(parent_row_number)
+            max_value = prerequisite_rows_to_dependent_rows[parent_row_number].max
+            prerequisite_row_map[row_counter] << max_value unless prerequisite_row_map[row_counter].include?(max_value)
+          else
+            # THIS row's prerequisite row has no other rows that are dependent on it,
+            # so we'll make THIS row directly dependent on it.
+            prerequisite_row_map[row_counter] << parent_row_number
+          end
+
+          # Keep track of the set of all rows that depend on this prerequisite parent row
+          prerequisite_rows_to_dependent_rows[parent_row_number] ||= []
+          prerequisite_rows_to_dependent_rows[parent_row_number] << row_counter
+        end
+      end
+    end
+    prerequisite_row_map
+  end
+
+  def self.identifiers_to_row_numbers_for_csv_data(csv_data_string)
+    identifiers_to_row_numbers = {}
+    indexes_of_identifier_headers = []
+
+    row_number_for_header_row = 0
+
+    CSV.parse(csv_data_string) do |row|
+      unless row[0].start_with?('_')
+        # Go to second row if first row doesn't contain headers
+        row_number_for_header_row += 1
+        next
+      end
+      indexes_of_identifier_headers = indexes_of_identifier_headers_for_row(row)
+      break
+    end
+
+    return {} if indexes_of_identifier_headers.blank?
+
+    row_counter = -1
+    CSV.parse(csv_data_string) do |row|
```

Comment:
there is a lot of syntax overhead from supporting the ignorable "human readable" label row, I think - otherwise we could just tell `CSV` to use the headers as keys.

Reply:
I am 10,000% in favor of getting rid of that "human readable" label row.  It's not nearly as useful as we thought it would be, and is actually more trouble than it's worth for the people who originally asked for it.

Reply:
@cgalarza I wish that worked, and I thought it would.  Unfortunately, it gives me this error:
```undefined method `with_index' for #<Array:0x007fd0f3a095d0>```

Reply:
actually, this works: ```CSV.parse(csv_data_string).each_with_index do |row, csv_row_number|```

But does that mean I'm technically calling ```each.each_with_index```?  I'm using CSV.parse in order to avoid keeping the entire CSV in memory and only working with one line at a time.  If I call ```each.each_with_index```, does that mean that I'm aggregating the entire set of results from each and keeping them in memory in order pass them to each_with_index?  @barmintor, do you have any thoughts about that too?